addons/mail/wizard/mail_compose_message.py

```diff
@@ -595,6 +595,10 @@ def _onchange_template_id(self, template_id, composition_mode, model, res_id):
             default_values = self.with_context(default_composition_mode=composition_mode, default_model=model, default_res_id=res_id).default_get(['composition_mode', 'model', 'res_id', 'parent_id', 'partner_ids', 'subject', 'body', 'email_from', 'reply_to', 'attachment_ids', 'mail_server_id'])
             values = dict((key, default_values[key]) for key in ['subject', 'body', 'partner_ids', 'email_from', 'reply_to', 'attachment_ids', 'mail_server_id'] if key in default_values)
 
+        if template_id:  # Restore default sender if not updated on template switch (for both "mass_mail" and "comment" modes)
+            if 'email_from' not in values.keys():
```

Comment:
Could be simplified, isn't it ?
```suggestion
            if 'email_from' not in values:
```